datahub-web-react/src/app/permissions/roles/ManageRoles.tsx

```diff
@@ -0,0 +1,262 @@
+`;
+
+const PageContainer = styled.span`
+    width: 100%;
+`;
+
+const DEFAULT_PAGE_SIZE = 10;
+
+// TODO: Cleanup the styling.
+export const ManageRoles = () => {
+    const entityRegistry = useEntityRegistry();
+    const location = useLocation();
+    const params = QueryString.parse(location.search, { arrayFormat: 'comma' });
+    const paramsQuery = (params?.query as string) || undefined;
+    const [query, setQuery] = useState<undefined | string>(undefined);
+    const [isBatchAddRolesModalVisible, setIsBatchAddRolesModalVisible] = useState(false);
+    const [focusRole, setFocusRole] = useState<DataHubRole>();
+    const [showViewRoleModal, setShowViewRoleModal] = useState(false);
+    useEffect(() => setQuery(paramsQuery), [paramsQuery]);
+
+    // Role list paging.
+    const [page, setPage] = useState(1);
+    const pageSize = DEFAULT_PAGE_SIZE;
+    const start = (page - 1) * pageSize;
+
+    const {
+        loading: rolesLoading,
+        error: rolesError,
+        data: rolesData,
+        refetch: rolesRefetch,
+    } = useListRolesQuery({
+        fetchPolicy: 'no-cache',
+        variables: {
+            input: {
+                start,
+                count: pageSize,
+                query,
+            },
+        },
+    });
+
+    const totalRoles = rolesData?.listRoles?.total || 0;
+    const roles = useMemo(() => rolesData?.listRoles?.roles || [], [rolesData]);
+    const onViewRole = (role: DataHubRole) => {
+        setFocusRole(role);
+        setShowViewRoleModal(true);
+    };
+    const resetRoleState = () => {
+        setIsBatchAddRolesModalVisible(false);
+        setShowViewRoleModal(false);
+        setFocusRole(undefined);
+    };
+
+    const [batchAssignRoleToActorsMutation] = useBatchAssignRoleToActorsMutation();
+    // eslint-disable-next-line
+    const batchAssignRoleToActors = (actorUrns: Array<string>) => {
+        if (!focusRole || !focusRole.urn) {
+            return;
+        }
+        batchAssignRoleToActorsMutation({
+            variables: {
+                input: {
+                    roleUrn: focusRole?.urn,
+                    actors: actorUrns,
+                },
+            },
+        })
+            .then(({ errors }) => {
+                if (!errors) {
+                    message.success({
+                        content: `Assigned Role to users!`,
+                        duration: 2,
+                    });
+                    setTimeout(function () {
+                        rolesRefetch();
+                    }, 3000);
+                }
+            })
+            .catch((e) => {
+                message.destroy();
+                message.error({ content: `Failed to assign Role to users: \n ${e.message || ''}`, duration: 3 });
+            })
+            .finally(() => {
+                resetRoleState();
+            });
+    };
+
+    const onChangePage = (newPage: number) => {
+        setPage(newPage);
+    };
+
+    const tableColumns = [
```

Comment:
Chris has a much prettier way he's been designing our Tables. Let's make sure to refactor this into a cleaner form as a followup cc @chriscollins3456

Reply:
Good to know!